hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/TarContainerPacker.java

```diff
@@ -84,11 +84,19 @@
         if (name.startsWith(DB_DIR_NAME + "/")) {
           Path destinationPath = dbRoot
               .resolve(name.substring(DB_DIR_NAME.length() + 1));
-          extractEntry(archiveInput, size, dbRoot, destinationPath);
+          if (entry.isDirectory()) {
+            Files.createDirectories(destinationPath);
+          } else {
+            extractFileEntry(archiveInput, size, dbRoot, destinationPath);
+          }
```

Comment:
Nit: can we move this logic to `extractEntry` (and keep the method name) to avoid duplication?

Reply:
done!